gateway/middleware.go

```diff
@@ -226,16 +228,22 @@ func (t BaseMiddleware) Config() (interface{}, error) {
 }
 
 func (t BaseMiddleware) OrgSession(orgID string) (user.SessionState, bool) {
-	// Try and get the session from the session store
-	session, found := t.Spec.OrgSessionManager.SessionDetail(orgID, orgID, false)
-	if found && t.Spec.GlobalConfig.EnforceOrgDataAge {
-		// If exists, assume it has been authorized and pass on
-		// We cache org expiry data
-		t.Logger().Debug("Setting data expiry: ", session.OrgID)
-		ExpiryCache.Set(session.OrgID, session.DataExpires, cache.DefaultExpiration)
+	var session user.SessionState
+	var found bool
+
+	if !rpc.IsEmergencyMode() {
```

Comment:
so in emergency mode session will always be nil. Won't that impact the gateway?

Reply:
It is not nil, but empty object, pls we return `found` as false so I guess we should be save. 

But I would rewrite it to the guard statement like this:
```
if rpc.IsEmergencyMode() {
   return session, false
}
```